cmp/grid/columns/Column.js

```diff
@@ -202,16 +213,21 @@ export class Column {
         this.colId = withDefault(colId, field);
         throwIf(!this.colId, 'Must specify colId or field for a Column.');
 
-        this.hidden = withDefault(hidden, false);
-        warnIf(rest.hide, `Column ${this.colId} configured with {hide: true} - use "hidden" instead.`);
+        this.isTreeColumn = withDefault(isTreeColumn, false);
+
+        // Note that parent GridModel might have already defaulted displayName from an associated
+        // `Store.field` when pre-processing Column configs - prior to calling this ctor.
+        this.displayName = withDefault(displayName, genDisplayName(this.colId));
```

Comment:
now might be a good time to actually rewrite all of these withDefault() statements to the more concise `??`.
The only time withDefault() is warranted is if we want to allow the application to force a setting of null or undefined.
(don't think that is the case here for any of these)

Reply:
It's a good question and I thought about it some more.

* I changed the `displayName` statement to ensure the default off of colId is always assigned. I agree I don't think we ever intend or want an app to force a null there. However, here and in some other cases I think we want `||` not `??` as we don't want to allow `''` or another false value here.
* The headerName is different - there you might well not want a header (for a minimal icon column, e.g.).  So we need something more than a falsely check and withDefault works nicely.
* The bool flags that default to false could go to `!!flagConfig` but I think withDefault ends up be more immediately readable and then is consistent with the flags we wish to default to true. Dev could still pass and end up with null, but that should be fine as a false value anyway.
* Reviewed one or two others - see a9780b43400bcc64c5b31aa68a7ec18adc64f61f